lib/web_ui/lib/ui.dart

```diff
@@ -50,3 +43,124 @@ const bool isWeb = true;
+  final Map<String, PlatformViewFactory> registeredFactories =
+      <String, PlatformViewFactory>{};
+
+  final Map<int, html.Element> _createdViews = <int, html.Element>{};
+
+  /// Private constructor so this class can be a singleton.
+  PlatformViewRegistry._();
+
+  /// Register [viewTypeId] as being creating by the given [factory].
+  bool registerViewFactory(String viewTypeId, PlatformViewFactory factory) {
+    if (registeredFactories.containsKey(viewTypeId)) {
+      return false;
+    }
+    registeredFactories[viewTypeId] = factory;
+    return true;
+  }
+
+  /// Returns the view that has been created with the given [id], or `null` if
+  /// no such view exists.
+  html.Element? getCreatedView(int id) {
+    return _createdViews[id];
+  }
+}
+
+/// A function which takes a unique [id] and creates an HTML element.
+typedef PlatformViewFactory = html.Element Function(int viewId);
+
+/// The platform view registry for this app.
+final PlatformViewRegistry platformViewRegistry = PlatformViewRegistry._();
+
+/// Handles a platform call to `flutter/platform_views`.
+///
+/// Used to create platform views.
+void handlePlatformViewCall(
+  ByteData data,
+  PlatformMessageResponseCallback callback,
+) {
+  const engine.MethodCodec codec = engine.StandardMethodCodec();
+  final engine.MethodCall decoded = codec.decodeMethodCall(data);
+
+  switch (decoded.method) {
+    case 'create':
+      _createPlatformView(decoded, callback);
+      return;
+    case 'dispose':
+      _disposePlatformView(decoded, callback);
+      return;
+  }
+  callback(null);
+}
+
+void _createPlatformView(
+    engine.MethodCall methodCall, PlatformMessageResponseCallback callback) {
+  final Map<dynamic, dynamic> args = methodCall.arguments;
+  final int id = args['id'];
+  final String viewType = args['viewType'];
+  const engine.MethodCodec codec = engine.StandardMethodCodec();
+
+  // TODO(het): Use 'direction', 'width', and 'height'.
+  final PlatformViewFactory? platformViewFactory = platformViewRegistry.registeredFactories[viewType];
+  if (platformViewFactory == null) {
+    callback(codec.encodeErrorEnvelope(
+      code: 'Unregistered factory',
+      message: "No factory registered for viewtype '$viewType'",
+    ));
+    return;
+  }
+  // TODO(het): Use creation parameters.
+  final html.Element element = platformViewFactory(id);
+
+  platformViewRegistry._createdViews[id] = element;
+  callback(codec.encodeSuccessEnvelope(null));
+}
+
+void _disposePlatformView(
+    engine.MethodCall methodCall, PlatformMessageResponseCallback callback) {
+  final int id = methodCall.arguments;
+  const engine.MethodCodec codec = engine.StandardMethodCodec();
+
+  // Remove the root element of the view from the DOM.
+  platformViewRegistry._createdViews[id]?.remove();
+  platformViewRegistry._createdViews.remove(id);
+
+  callback(codec.encodeSuccessEnvelope(null));
+}
+
+// TODO(yjbanov): remove _Callback, _Callbacker, and _futurize. They are here only
```

Comment:
I think this bug just got fixed in master, but probably hasn't rolled to you yet.